src/fdb5/api/helpers/Callback.h

```diff
@@ -21,13 +21,30 @@
 namespace fdb5 {
 
 class FDB;
+class CallbackInterface;
 
 using ArchiveCallback = std::function<void(const Key& key, const void* data, size_t length, std::future<std::shared_ptr<const FieldLocation>>)>;
 using FlushCallback = std::function<void()>;
-using ConstructorCallback = std::function<void(FDB&)>;
+using ConstructorCallback = std::function<void(CallbackInterface&)>;
 
-static const ArchiveCallback CALLBACK_NOOP = [](const Key& key, const void* data, size_t length, std::future<std::shared_ptr<const FieldLocation>>) {};
+static const ArchiveCallback CALLBACK_ARCHIVE_NOOP = [](const Key& key, const void* data, size_t length, std::future<std::shared_ptr<const FieldLocation>>) {};
 static const FlushCallback CALLBACK_FLUSH_NOOP = []() {};
-static const ConstructorCallback CALLBACK_CONSTRUCTOR_NOOP = [](FDB&) {};
+static const ConstructorCallback CALLBACK_CONSTRUCTOR_NOOP = [](CallbackInterface&) {};
```

Comment:
You can omit the types in the parameters or the `NOOP` CBs, e.g.

```c++
static const ArchiveCallback CALLBACK_ARCHIVE_NOOP = [](auto&&, auto&&, auto&&, auto&&) {};
 static const FlushCallback CALLBACK_FLUSH_NOOP = [](){};
 static const ConstructorCallback CALLBACK_CONSTRUCTOR_NOOP = [](auto&&) {};
```

Note you need to use `auto&&` to prevent introducing a copy for types passed as references when calling the `NOOP`